test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1777,9 +1823,8 @@ def test_cgroup_cpuset_ncpus_are_cores(self):
                     cores = int(cores_match.groups()[0])
                 if phys_match:
                     pval = int(phys_match.groups()[0])
-                    if pval != prev:
-                        prev = pval
-                        phys += 1
```

Comment:
This has already been fixed in https://github.com/PBSPro/pbspro/pull/1709, so best to just rebase master and pick that up.

Reply:
That test has other problems -- so I'm sticking to my guns for the moment:

-on TH3 the timing introduces a spurious failure -- 10 attempts are not enough to make all jobs that run actually run; the fact the daemons and hooks are CPU starved and that the exechost_periodic interlopes makes it unsafe. Better sleep a short time between jobs...and increase the number of attempts.
-The job runs are so slow that a default sleep of 100 is too dangerous on large machines -- by the time you get to the last job the first ones will have finished.
-We're submitting far too many jobs for what the test tries to test. As a result, on large machines we'll just skip (with a "heuristic" limit of 64 jobs).
-The test at the end is also not safe. It "expects" a queued job, but since it looks just after the job is submitted the scheduler might not yet have gotten to it, which would produce false positives. If the job starts Q and then becomes R, "expect" will not do 10 attempts and then fail when the job becomes "R", it will just declare victory at the first Q.

So I'm making a version that will first submit one job per phys, of "cores" width. That should make it practical even on large _and_ slow machines. I'm setting the sleep to 3600 to avoid spurious "success" by jobs finishing early, and I'm _first_ sleeping for 30 seconds before doing an expect of state Q for the job that should not run (which I keep as a 1 CPU job).

Reply:
In other words: I'll rebase, but I still plan to keep my test in the merge ;-).

Reply:
Rebased. Did keep a set instead of a dictionary with separate keys and constant all-one values, since they are functionally equivalent but the set is more readable and natural.